src/models/Frame.cpp

```diff
@@ -4,56 +4,115 @@
 
 #include <iostream>
 
+Frame::Frame(const float *depthMap, const BYTE *colorMap,
+             const Eigen::Matrix3f &depthIntrinsics,
+             const Eigen::Matrix4f &depthExtrinsicsInv,
+             const Eigen::Matrix4f &trajectoryInv, int depthWidth,
+             int depthHeight) {
+  computeVertexMap(depthMap, depthIntrinsics, depthWidth, depthHeight);
+  computeNormalMap(depthWidth, depthHeight);
+  setExtrinsicMatrix(depthExtrinsicsInv.inverse());
+  this->depthWidth = depthWidth;
+  this->depthHeight = depthHeight;
+  this->intrinsicMatrix = depthIntrinsics;
+  std::cout << "Frame created!" << std::endl;
+}
+
+bool Frame::containsImgPoint(Eigen::Vector2i imgPoint) {
+  return imgPoint[0] >= 0 && imgPoint[0] < depthWidth && imgPoint[1] >= 0 &&
+         imgPoint[1] < depthHeight;
+}
+
+int Frame::getFrameHeight() { return depthHeight; }
+
+int Frame::getFrameWidth() { return depthWidth; }
+
+std::vector<Eigen::Vector3f> &Frame::getVertexMap() { return mVertices; }
+std::vector<Eigen::Vector3f> &Frame::getNormalMap() { return mNormals; }
+
+std::vector<Eigen::Vector3f> &Frame::getVertexMapGlobal() {
+  return mVerticesGlobal;
+}
+
+Eigen::Vector3f Frame::projectPointIntoFrame(const Eigen::Vector3f &point) {
+  Eigen::Matrix4f extrinsicMatrixInverse = extrinsicMatrix.inverse();
+  // std::cout << extrinsicMatrixInverse << std::endl;
+  const auto rotation = extrinsicMatrixInverse.block(0, 0, 3, 3);
+  const auto translation = extrinsicMatrixInverse.block(0, 3, 3, 1);
+  return rotation * point + translation;
+}
```

Comment:
```suggestion
Eigen::Vector3f Frame::projectPointIntoFrame(const Eigen::Vector3f &point) {
  // std::cout << extrinsicMatrixInverse << std::endl;
  const auto rotation = extrinsicMatrix.block(0, 0, 3, 3);
  const auto translation = extrinsicMatrix.block(0, 3, 3, 1);
  return rotation * point + translation;
}
```
Ths helper function is used to transform point from global coordinate to frame coordinate. I guess extrinsic matrix is the one to be used and not its inverse.

Reply:
Yes, I agree.